src/Cardano/Signer.cpp

```diff
@@ -241,10 +297,12 @@ std::vector<TxInput> Signer::selectInputsWithTokens(const std::vector<TxInput>&
 }
 
 // Create a simple plan, used for estimation
-TransactionPlan simplePlan(Amount amount, const TokenBundle& requestedTokens, const std::vector<TxInput>& selectedInputs, bool maxAmount) {
+TransactionPlan simplePlan(Amount amount, const TokenBundle& requestedTokens, const std::vector<TxInput>& selectedInputs, bool maxAmount, uint64_t deposit, uint64_t undeposit) {
     TransactionPlan plan;
     plan.amount = amount;
     plan.utxos = selectedInputs;
+    plan.deposit = deposit;
+    plan.undeposit = undeposit;
```

Comment:
here we can take advantage of C++20 aggregate initializer:

```cpp
TransactionPlan{.field = my_value, .field2 = another_value}
```

and respect the RAII idiom pattern